bodo/transforms/distributed_analysis_call_registry.py

```diff
@@ -0,0 +1,634 @@
+"""
+Provides a registry of function call handlers for distributed analysis.
+"""
+
+from numba.core import types
+from numba.core.ir_utils import guard
+
+from bodo.transforms.distributed_analysis import (
+    Distribution,
+    _meet_array_dists,
+    _set_REP,
+    _set_var_dist,
+)
+from bodo.utils.typing import BodoError, get_overload_const_str, is_overload_none
+from bodo.utils.utils import is_distributable_typ
+
+
+class DistributedAnalysisContext:
+    """Distributed analysis context data needed for handling calls"""
+
+    def __init__(self, typemap, array_dists, equiv_set, func_name, metadata, diag_info):
+        self.typemap = typemap
+        self.array_dists = array_dists
+        self.equiv_set = equiv_set
+        self.func_name = func_name
+        self.metadata = metadata
+        self.diag_info = diag_info
+
+
+class DistributedAnalysisCallRegistry:
+    """Registry of function call handlers for distributed analysis"""
+
+    def __init__(self):
+        self.call_map = {
```

Comment:
Please add description of the expected contents of each tuple.

Reply:
Done.